adaptors_task/student/include/impl.h

```diff
@@ -11,4 +12,38 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source) {
+    std::stack<char> stack;
+    for (auto const &c : source) {
+        if (c == '(' || c == '[' || c == '{') {
```

Comment:
It's better to pack this into some container

Reply:
I did it and the code is smaller now and can be extended much more simple.